COORDINATE_SYSTEM2/COORDINATE_SYSTEM2/CoordinateSystem.cs

```diff
@@ -6,28 +6,112 @@
 
 namespace COORDINATE_SYSTEM2
 {
+    /// <summary>
+    /// Клас Координатна система.
+    /// Містить Ліст всіх створених точок та Ліст всіх знайдених дистанцій.
+    /// Містить Add(), який створює нову точку, добавляє її в Ліст, шукає відстані 
+    /// між існуючими точками до нової точки і добавляє їх в ліст.
+    /// Містить DisplayAllPoints(), який друкує всі точки з ліста.
+    /// Містить MaxDistance() для пошуку максимальної відстані між точками.
+    /// Містить PrintAllDistances(), який друкує всі наявні відстані з Ліста.
+    /// </summary>
     internal class CoordinateSystem
     {
-        List<Point> listOfPoints = new List<Point>();
+        List<IFigure> listOfFigures = new List<IFigure>();
 
         List<PointsDistance> pointsDistances = new List<PointsDistance>();
 
-        public void Add(Point point)
+        //TODO мені потрібно зробити КЛАС!!! "ФІГУРА"
+        //- який буде базовим для Точки, Прямокутника та Трикутника.
+        //І в методі Ед приймати їх через цей базовий клас.
+        public void Add(IFigure figure) 
         {
-            foreach (Point point1 in listOfPoints)
+            if(figure.Area() == 0)
```

Comment:
why do you even need this dumb `if`?)

Reply:
why do you even need to check Area on adding the figure into CoordinateSystem)0)